### DISCUSSION TITLE
[Feature]: Support transpiling external JSX

### DISCUSSION THREAD

**Discussion by jaschaio** [remix-run/remix#3098] [Proposals]
Coming from this issue #1423 I am trying to use a common components module in a monorepo.

`serverDependenciesToBundle` works, but I can't use jsx within those local modules, as I get the Following error:

```
SyntaxError: Unexpected token '<'
```

Any way I can transpile them using JSX?

**Comment by jaschaio** [comment]
Ok, two hours later I found out that what actually happens is that `serverDependenciesToBundle` expects an array of regex patterns instead of strings as it did earlier.

**Reply by MiguelNiblock** [reply]
@jaschaio Did you find a solution to this problem? I would like to know if I can transpile UI component libraries that come from react-native which are also supposed to work on web, but they assume you will use babel. For example native-base or react-native-paper

**Reply by jaschaio** [reply]
@MiguelNiblock I was not able to make this work because of #3102
